kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/ActionsIterator.java

```diff
@@ -45,17 +44,19 @@ class ActionsIterator implements CloseableIterator<ActionWrapper> {
     private final TableClient tableClient;
 
     /**
-     * Iterator over the files.
+     * Linked list of iterator files (commit files and/or checkpoint files)
+     * {@link LinkedList} to allow removing the head of the list and also to peek at the head
+     * of the list. The {@link Iterator} doesn't provide a way to peek.
      * <p>
-     * Each file will be split (by 1, or more) to yield an iterator of FileDataReadResults.
+     * Each of these files return an iterator of {@link ColumnarBatch} containing the actions
      */
-    private final Iterator<FileStatus> filesIter;
+    private final LinkedList<FileStatus> filesList;
```

Comment:
I think the downside of this is slightly higher memory usage. Is there an estimate of that? What will be the size of this iterator in the worst case? 

I guess the worst case will be for a table that has a 1000s of commits in the log, and has no _last_checkpoint file? Will this list be populated with the log segment to read, or the entire log directory list?

Reply:
sounds good.